src/System Application/App/Page Summary Provider/src/PageSummaryProviderImpl.Codeunit.al

```diff
@@ -117,16 +122,47 @@ codeunit 2717 "Page Summary Provider Impl."
         if Handled then begin // Partner overrode fields
             Session.LogMessage('0000D73', StrSubstNo(OnBeforeGetPageSummaryWasHandledTxt, PageId), Verbosity::Normal, DataClassification::SystemMetadata, TelemetryScope::ExtensionPublisher, 'Category', PageSummaryCategoryLbl);
             AddFieldsToResult(FieldsJsonArray, ResultJsonObject);
-            exit(Format(ResultJsonObject));
+            exit;
         end;
 
         // Get summary fields
-        if not TryGetPageSummaryFields(PageId, RecId, Bookmark, ResultJsonObject) then begin
+        if not TryGetPageSummaryFields(PageId, RecId, Bookmark, ResultJsonObject) then
             AddErrorMessage(ResultJsonObject, FailedGetSummaryFieldsCodeTok, GetLastErrorText());
-            exit(Format(ResultJsonObject));
+    end;
+
+    local procedure GetRecordFields(PageId: Integer; Bookmark: Text; var ResultJsonObject: JsonObject)
+    begin
+        // Get all visible and available table fields that back the controls that are visible on the page
+        if TryGetAvailableRecordFieldsData(PageId, Bookmark, ResultJsonObject) then;
```

Comment:
Who do we expect to throw an `Error`? Should we add a `Session.LogMessage`/warning here?